lib/portlayer/storage/image_cache.go

```diff
@@ -300,3 +298,54 @@ func (c *NameLookupCache) ListImages(ctx context.Context, store *url.URL, IDs []
 
 	return imageList, nil
 }
+
+// DeleteImage deletes an image from the image store.  If it is in use or is being inheritted from, then this will return an error.
+func (c *NameLookupCache) DeleteImage(ctx context.Context, image *Image) error {
+	infof("DeleteImage: deleting %s", image.Self())
+
+	// Check the image exists.  This will rehydrate the cache if necessary.
+	_, err := c.GetImage(ctx, image.Store, image.ID)
+	if err != nil {
+		return errorf("DeleteImage: %s", err)
```

Comment:
If `err` is ErrNotExist, should we be preserving that so callers can map errors to detailed responses.